src/libstd/path.rs

```diff
@@ -1419,20 +1423,28 @@ impl<'a, T: ?Sized + AsRef<OsStr>> From<&'a T> for PathBuf {
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl From<OsString> for PathBuf {
+    /// Converts a `OsString` into a `PathBuf`.
+    /// This conversion copies the data.
+    /// This conversion does allocate memory.
     fn from(s: OsString) -> PathBuf {
         PathBuf { inner: s }
     }
 }
 
 #[stable(feature = "from_path_buf_for_os_string", since = "1.14.0")]
 impl From<PathBuf> for OsString {
+    /// Converts a `PathBuf` into a `OsString`.
+    /// This conversion copies the data.
+    /// This conversion does allocate memory.
```

Comment:
Same here, but the other way around: Unwrapping a wrapper is free!

https://godbolt.org/g/Aoe8Q1